storagemarket/impl/client.go

```diff
@@ -200,22 +202,23 @@ func (c *Client) Start(ctx context.Context, p ClientDealProposal) (cid.Cid, erro
 		return cid.Undef, xerrors.Errorf("computing commP failed: %w", err)
 	}
 
-	dealProposal := &storagemarket.StorageDealProposal{
-		PieceRef:             commP,
-		PieceSize:            uint64(pieceSize),
+	dealProposal := market.DealProposal{
+		PieceCID:             commcid.PieceCommitmentV1ToCID(commP),
```

Comment:
Does it make sense to have `PieceIO.GeneratePieceCommitment` return a `cid.CID`?  Would remove a little boilerplate from client and provider.

Reply:
hmmm... good question. Let's leave it for a future discussion